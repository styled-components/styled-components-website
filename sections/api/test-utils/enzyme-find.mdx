### `enzymeFind` | v4

A convenience method for finding instances of a particular styled component within an enzyme wrapper.

```jsx
import { mount } from 'enzyme'
import styled from 'styled-components'
import { enzymeFind } from 'styled-components/test-utils'

const Foo = styled.div`
  color: red;
`

const wrapper = mount(
  <div>
    <Foo>bar</Foo>
  </div>
)

enzymeFind(wrapper, Foo)
```
