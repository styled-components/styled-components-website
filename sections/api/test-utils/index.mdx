import Find from './find.mdx'
import FindAll from './find-all.mdx'
import EnzymeFind from './enzyme-find.mdx'

## Test Utilities

<Find />

<FindAll />

<EnzymeFind />
