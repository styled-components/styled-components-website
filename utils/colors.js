import lighten from 'polished/lib/color/lighten';
import darken from 'polished/lib/color/darken';
import shade from 'polished/lib/color/shade';

export const paleGrey = '#efefef';
export const lightGrey = 'rgba(20, 20, 20, 0.1)';
export const darkGrey = darken(0.05, '#282a36');
export const grey = '#282a36';

export const red = '#ff5555';
export const violetRed = 'rgb(219, 112, 147)';
export const darkVioletRed = darken(0.1, 'rgb(219, 112, 147)');
export const lightVioletRed = lighten(0.31, 'rgb(219, 112, 147)');

export const gold = shade(0.9, 'rgb(243, 182, 97)');
