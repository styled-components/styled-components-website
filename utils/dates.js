export const getFormattedDate = (unformattedDate) => {
  const date = new Date(unformattedDate);
  const year = date.getFullYear();
  let month = (1 + date.getMonth()).toString();
  month = month.length > 1 ? month : '0' + month;
  let day = date.getDate().toString();
  day = day.length > 1 ? day : '0' + day;

  return day + '/' + month + '/' + year;
};
