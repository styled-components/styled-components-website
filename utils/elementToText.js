import { isValidElement } from 'react';

const whitespacesRe = /\s+/g;
const _format = (str = '') => str.trim().replace(whitespacesRe, ' ');

const elementToTextRec = (x) => {
  if (Array.isArray(x)) {
    return x.map(elementToTextRec).join('');
  } else if (isValidElement(x)) {
    return elementToTextRec(x.children || x.props.children);
  } else if (typeof x === 'string') {
    return x || '';
  }

  return '';
};

const elementToText = (node) => _format(elementToTextRec(node));

export default elementToText;
