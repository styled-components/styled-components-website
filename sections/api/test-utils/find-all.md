### `findAll` | v3

A convenience method to find all instances of a styled component's rendered DOM node within a given DOM root.

```js
import styled from 'styled-components'
import { findAll } from 'styled-components/test-utils'

const Foo = styled.div`
  color: ${props => props.color};
`

/**
 * Somewhere in your app:
 *
 * ReactDOM.render(
 *   <main>
 *     <Foo color="red" />
 *     <Foo color="green" />
 *   </main>, document.body
 * );
 */

// retrieves a NodeList of instances of "Foo" in the body (querySelectorAll under the hood)
findAll(document.body, Foo) // NodeList<HTMLDivElement> | null
```
