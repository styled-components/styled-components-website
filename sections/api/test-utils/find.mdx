### `find` | v3

A convenience method to find a single instance of a styled component's rendered DOM node within a given DOM root.

```js
import styled from 'styled-components'
import { find } from 'styled-components/test-utils'

const Foo = styled.div`
  color: red;
`

/**
 * Somewhere in your app:
 *
 * ReactDOM.render(
 *   <main>
 *     <Foo />
 *   </main>, document.body
 * );
 */

// retrieves the first instance of "Foo" in the body (querySelector under the hood)
find(document.body, Foo) // HTMLDivElement | null
```
