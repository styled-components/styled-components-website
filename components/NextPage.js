import styled, { css } from 'styled-components';
import { KeyboardArrowRight } from '@styled-icons/material';

import rem from '../utils/rem';
import Link from './Link';
import { grey } from '../utils/colors';
import { mobile } from '../utils/media';
import { headerFont } from '../utils/fonts';

const Wrapper = styled(Link).attrs((/* props */) => ({
  unstyled: true,
}))`
  display: flex;
  flex-direction: row;
  align-items: stretch;
  justify-content: flex-end;

  width: 100%;
  padding: ${rem(40)} ${rem(20)};
  text-align: right;
  font-family: ${headerFont};

  ${mobile(css`
    text-align: left;
    justify-content: center;
    padding: ${rem(30)} ${rem(20)};
  `)};
`;

const Text = styled.h3`
  font-weight: normal;
  padding-right: ${rem(20)};
  margin: 0;
`;

const PageName = styled.h2`
  font-weight: 600;
  padding-right: ${rem(20)};
  margin: 0;
`;

const Icon = styled(KeyboardArrowRight)`
  color: ${grey};
  width: ${rem(30)};
`;

const NextPage = ({ title, href }) => (
  <Wrapper unstyled href={href}>
    <div>
      <Text>Continue on the next page</Text>
      <PageName>{title}</PageName>
    </div>

    <div>
      <Icon />
    </div>
  </Wrapper>
);

export default NextPage;
