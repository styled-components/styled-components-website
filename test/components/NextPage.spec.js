import React from 'react';
import renderer from 'react-test-renderer';
import { mockRouter } from '../../utils/test-utils';
import NextPage from '../../components/NextPage';

mockRouter();

test('NextPage renders correctly', () => {
  const tree = renderer.create(<NextPage href="" />).toJSON();

  expect(tree).toMatchSnapshot();
});
