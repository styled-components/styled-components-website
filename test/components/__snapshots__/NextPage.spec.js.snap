// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextPage renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  width: 100%;
  padding: 2.2222222222222223rem 1.1111111111111112rem;
  text-align: right;
  font-family: "Avenir Next",-apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
}

.c1 {
  font-weight: normal;
  padding-right: 1.1111111111111112rem;
  margin: 0;
}

.c2 {
  font-weight: 600;
  padding-right: 1.1111111111111112rem;
  margin: 0;
}

.c3 {
  color: #282a36;
  width: 1.6666666666666667rem;
}

@media (max-width:62.5em) {
  .c0 {
    text-align: left;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    padding: 1.6666666666666667rem 1.1111111111111112rem;
  }
}

<a
  className="c0"
  href=""
>
  <div>
    <h3
      className="c1"
    >
      Continue on the next page
    </h3>
    <h2
      className="c2"
    />
  </div>
  <div>
    <svg
      className="c3"
    />
  </div>
</a>
`;
